spec/support/jazz.rb

```diff
@@ -319,6 +319,14 @@ def self.resolve_type(object, context)
         end
       end
     end
+
+    def self.filter_possible_types(types, ctx)
+      types = types.map { |type| GraphQL::BaseType.resolve_related_type(type) }
```

Comment:
This `GraphQL::BaseType.resolve_related_type(type)` is a bit of a pain, I think this is a code smell and should be fixed higher up so the client doesn't have to deal with this themselves. But leaving it here now as a proof of concept.